flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -776,6 +780,14 @@ protected void convertSelectImpl(final Blackboard bb, SqlSelect select) {
         convertOrder(select, bb, collation, orderExprList, select.getOffset(), select.getFetch());
 
         if (select.hasHints()) {
+            FlinkContext context = ShortcutUtils.unwrapContext(cluster);
+            if (context.getTableConfig().get(TableConfigOptions.TABLE_QUERY_HINTS_IGNORE)) {
+                throw new ValidationException(
```

Comment:
Thanks for working on this topic. I was wondering why exception will be thrown. The idea was to ingore hints, not disallow them.

Reply:
+1 otherwise such change leads to breaking behaviour and i guess making it violating SQL Standard saying that content of comments should not have impact on query output

Reply:
My assumption was also regarding the naming apologize if I misinterpreted it incorrectly.
Currently, a ValidationException is thrown when a user specifies an OPTIONS hint if the OPTIONS hint is disabled.